during_workshop_7.md

```diff
@@ -7,96 +7,159 @@ Select the correct set of instructions for your platform of choice:
+TODO: Check how this changes if one person creates & registers on branch A, then second person on branch B
 
-It produces two reports: one in HTML form that you can open in your browser (DotnetTemplate.Web/coverage/index.html) and one in XML that we will get Jenkins to parse.
+To create your pipeline:
+* Click "Create Pipeline"
+* Select "Azure Repos Git" as your code location.
+* Select your repository
+* It should then locate your pipeline file, and you can click "Run"
 
-Try adding code coverage to Jenkins:
+After a few seconds, you should see a successful run! Check that it matches your repository & branch name, and then click into your pipeline & job to see the console output.
 
-1. Install the [Code Coverage API](https://plugins.jenkins.io/code-coverage-api/) plugin on Jenkins.
-2. Change your Jenkins pipeline to run the tests with code coverage.
-3. Add a step after running the tests to publish coverage. You can see a simple example of the command if you scroll down the Code Coverage API documentation, to the "pipeline example". You will want to use the "istanbulCoberturaAdapter", and the report to publish is "cobertura-coverage.xml" in the coverage folder.
-4. You should see a code coverage report appear on the build's page after it completes. Click through to see details.
+Can you see where your `echo` statements have printed their messages?
 
-Now let's enforce high code coverage:
+### Extending the pipeline
 
-1. Configure it to fail the build for code coverage below 90%. You may find it easiest to use the Jenkins [Snippet Generator](https://www.jenkins.io/doc/book/pipeline/getting-started/#snippet-generator).
-2. Push your change and watch the build go red!
-3. Edit the `DotnetTemplate.Web/Scripts/spec/exampleSpec.ts` file:
-  * Update the import statement: `import { functionOne, functionTwo } from '../home/example';`
-  * Invoke functionTwo on a new line in the test `functionTwo();`
-4. Push the change and observe the build go green again! You can also view the code coverage history.
+As before, we want our pipeline to be able to verify that our application is in a working state!
+
+Try extending your pipeline to cover all the scenarios so that it:
+1. Builds the C# code.
+1. Runs the C# tests.
+1. Builds the TypeScript code.
+1. Runs the linter on the TypeScript code.
+1. Runs the TypeScript tests.
+
+It's up to you how exactly you do this, but consider:
+* Can you install the appropriate version of DotNet and Node using [the provided tasks](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/?view=azure-pipelines)
+* Do you want to use scripts or tasks for the other steps?
+* Check the [list of options](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-script?view=azure-pipelines) on the `script` task - are there any that we should set for any of your steps?
+* Do you need the C# & TypeScript steps to run on the same agent?
+  * Could we parallelise them?
+  * Should we?
+
+### Stretch
```

Comment:
Idea from Alex for a bonus stretch goal - get the pipeline to raise a bug ticket if it fails